include/ignition/math/Line3.hh

```diff
@@ -229,6 +229,32 @@ namespace ignition
         return true;
       }
 
+      /// \brief Calculate shortest distance between line and point
+      /// \param[in] _pt Point which we are measuring distance to.
+      /// \returns Distance from point to line.
+      public: T Distance(const Vector3<T> &_pt)
+      {
+        auto line = this->pts[1] - this->pts[0];
+        auto ptTo0 = _pt - this->pts[0];
+        auto ptTo1 = _pt - this->pts[1];
+
+        // Point is projected beyond pt0
+        if (ptTo0.Dot(line) <= 0.0)
+        {
+          return ptTo0.Length();
+        }
+
+        // Point is projected beyond pt1
+        if (ptTo1.Dot(line) >= 0.0)
+        {
+          return ptTo1.Length();
+        }
+
+        // Distance to point projected onto line
+        auto d = (ptTo0).Cross(line);
+        return d.Length() / (line).Length();
```

Comment:
the other APIs in this class accept an `_epsilon` parameter with a default value of `1e-6`. I think we should do something similar to avoid the possibility of dividing by `0` here

Reply:
Resolved in 72d7779.